src/diff/index.js

```diff
@@ -241,11 +241,19 @@ export function diff(
 		newVNode._original = null;
 		// if hydrating or creating initial tree, bailout preserves DOM:
 		if (isHydrating || excessDomChildren != null) {
-			newVNode._dom = oldDom;
-			newVNode._hydrating = !!isHydrating;
-			excessDomChildren[excessDomChildren.indexOf(oldDom)] = null;
-			// ^ could possibly be simplified to:
-			// excessDomChildren.length = 0;
+			if (e.then) {
```

Comment:
This logic should only kick in when we are in a boundary, this means that only if we notice a promise is thrown up we should attempt at progressively hydrating this boundary.

Reply:
Hmm - elsewhere we use component._dirty to trigger this - it's valid to throw a non-Promise and catch it, and that should trigger progressive hydration.

Reply:
I mean, an error shouldn't hold progressive hydration. An error boundary will then trigger bugs like #2539